### PR TITLE
Fix react+uirevision when removing traces

### PR DESCRIPTION
**Pull request by etpinard**
fixes https://github.com/plotly/plotly.js/issues/3525

cc @plotly/plotly_js

### CODE REVIEW

**Comment by etpinard** [comment]
@antoinerg @archmoj  looking for reviewers on this one

I'll like to make another patch release this week before 1.45.0.

File: src/plot_api/plot_api.js [hunk 1]
```diff
@@ -2539,7 +2539,7 @@ function getTraceIndexFromUid(uid, data, tracei) {
         if(data[i].uid === uid) return i;
     }
     // fall back on trace order, but only if user didn't provide a uid for that trace
-    return data[tracei].uid ? -1 : tracei;
+    return (!data[tracei] || data[tracei].uid) ? -1 : tracei;
```
Comment (archmoj) [inline src/plot_api/plot_api.js]: Wondering if we may add few more checks to test `tracei` to be a valid index?
<pre>
return (tracei === undefined || tracei < 0 || !data[tracei] || data[tracei].uid) ? -1 : tracei;
</pre>

Comment (etpinard) [reply src/plot_api/plot_api.js]: `getTraceIndexFromUid` gets called here:

https://github.com/plotly/plotly.js/blob/7d117741d0e93f79831b65faa78a113bf3d8109b/src/plot_api/plot_api.js#L2628

and `index` is filled here:

https://github.com/plotly/plotly.js/blob/7d117741d0e93f79831b65faa78a113bf3d8109b/src/plots/plots.js#L1019

so I can't be anything other than positive number.

**Comment by archmoj** [comment]
:dancer:
@etpinard Thanks for the fix.

**Pull request merged**